replication.go

```diff
@@ -206,9 +206,8 @@ func (r *Raft) replicateTo(s *followerReplication, lastIndex uint64) (shouldStop
 START:
 	// Prevent an excessive retry rate on errors
 	if s.failures > 0 {
-		select {
-		case <-time.After(backoff(failureWait, s.failures, maxFailureScale)):
-		case <-r.shutdownCh:
+		if isTerminated(r.shutdownCh, backoff(failureWait, s.failures, maxFailureScale)) {
+			return
```

Comment:
this is a bug when received from `shutdownCh` but does not return. This PR also fix this